indico/core/auth.py

```diff
@@ -56,6 +56,18 @@ def synced_fields(self):
         synced_fields &= set(current_app.config['MULTIPASS_IDENTITY_INFO_KEYS'])
         return synced_fields
 
+    @property
+    def locked_fields(self):
+        """The keys that cannot be desynchronized.
+
+        This is a subset of the keys listed in ``synced_fields`` that cannot be
+        desynchronized by a non-admin user.
+        """
+        provider = self.sync_provider
+        if provider is None:
+            return set()
+        return set(provider.settings.get('locked_fields')) & self.synced_fields
```

Comment:
This fails if the setting isn't present

```suggestion
        return set(provider.settings.get('locked_fields', ())) & self.synced_fields
```